pkg/pods/pods_test.go

```diff
@@ -16,32 +19,191 @@ var (
 
 func TestPods_Create(t *testing.T) {
 	t.Parallel()
-	// TODO Figure out the rest.Config
-	fixture := New(fake.NewSimpleClientset(
-		testutils.NewPod("existing", testNamespace),
-	), nil, metav1.ListOptions{}, nil)
+	type TestCase struct {
+		test        string
+		name        string
+		namespace   string
+		status      string
+		delay       time.Duration
+		expectError bool
+		wait        string
+	}
 
-	options := PodOptions{
-		Name:          testName,
-		Namespace:     testNamespace,
-		Image:         "busybox",
-		Command:       []string{"sh", "-c", "sleep 300"},
-		RestartPolicy: k8sTypes.RestartPolicyNever,
+	testCases := []TestCase{
+		{
+			test:        "create pod not waiting",
+			name:        "pod-running",
+			namespace:   testNamespace,
+			status:      "Running",
+			delay:       1 * time.Second,
+			expectError: false,
+			wait:        "",
+		},
+		{
+			test:        "create failed pod not waiting",
+			name:        "pod-running",
+			namespace:   testNamespace,
+			status:      "Failed",
+			delay:       1 * time.Second,
+			expectError: false,
+			wait:        "",
+		}, {
+			test:        "wait for pod running",
+			name:        "pod-running",
+			namespace:   testNamespace,
+			status:      "Running",
+			delay:       1 * time.Second,
+			expectError: false,
+			wait:        "2s",
+		},
+		{
+			test:        "timeout waiting pod running",
+			name:        "pod-running",
+			namespace:   testNamespace,
+			status:      "Running",
+			delay:       10 * time.Second,
+			expectError: true,
+			wait:        "2s",
+		},
+		{
+			test:        "wait failed pod",
+			name:        "pod-running",
+			namespace:   testNamespace,
+			status:      "Failed",
+			delay:       1 * time.Second,
+			expectError: true,
+			wait:        "2s",
+		},
 	}
-	result, err := fixture.Create(options)
+	for _, tc := range testCases {
+		t.Run(tc.test, func(t *testing.T) {
+			// TODO Figure out the rest.Config
+			client := fake.NewSimpleClientset()
+			watcher := watch.NewFake()
+			client.PrependWatchReactor("pods", k8stest.DefaultWatchReactor(watcher, nil))
+			fixture := New(client, nil, metav1.ListOptions{}, nil)
+			go func(tc TestCase) {
+				time.Sleep(tc.delay)
+				watcher.Modify(testutils.NewPodWithStatus(tc.name, tc.namespace, tc.status))
+			}(tc)
 
-	if err != nil {
-		t.Errorf("encountered an error: %v", err)
-		return
+			result, err := fixture.Create(PodOptions{
+				Name:          tc.name,
+				Namespace:     tc.namespace,
+				Image:         "busybox",
+				Command:       []string{"sh", "-c", "sleep 300"},
+				RestartPolicy: k8sTypes.RestartPolicyNever,
+				Wait:          tc.wait,
+			})
+
+			if !tc.expectError && err != nil {
+				t.Errorf("unexpected error: %v", err)
+				return
+			}
+			if tc.expectError && err == nil {
+				t.Errorf("Expected an error but none returned")
```

Comment:
Minor nitpicks for any subsequent update: 
* static string can use `t.Error("...")`
* the error text should start with lower-case, i.e. `expected` vs. `Expected` as this would trigger a linting error once enabled

(This issue occurs in multiple places)